sdl_android/src/main/java/com/smartdevicelink/api/audio/AudioStreamManager.java

```diff
@@ -0,0 +1,192 @@
+import android.support.annotation.RequiresApi;
+import android.support.annotation.StringDef;
+import android.util.Log;
+
+import com.smartdevicelink.SdlConnection.SdlSession;
+import com.smartdevicelink.protocol.enums.SessionType;
+import com.smartdevicelink.proxy.interfaces.IAudioStreamListener;
+import com.smartdevicelink.proxy.interfaces.ISdl;
+import com.smartdevicelink.proxy.interfaces.ISdlServiceListener;
+import com.smartdevicelink.proxy.rpc.enums.BitsPerSample;
+import com.smartdevicelink.proxy.rpc.enums.SamplingRate;
+
+import java.io.File;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.LinkedList;
+import java.util.Queue;
+
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public class AudioStreamManager implements ISdlServiceListener {
+    private static final String TAG = AudioStreamManager.class.getSimpleName();
+
+    private ISdl sdlInterface;
+    private IAudioStreamListener sdlAudioStream;
+    private int sdlSampleRate;
+    private @SampleType int sdlSampleType;
+    private final Queue<BaseAudioDecoder> queue;
+    private boolean didRequestShutdown = false;
+
+    public AudioStreamManager(@NonNull ISdl sdlInterface, @NonNull @SamplingRate String sampleRate, @NonNull @BitsPerSample int sampleType) {
+        this.sdlInterface = sdlInterface;
+        this.queue = new LinkedList<>();
+
+        switch (sampleRate) {
+            case SamplingRate.EIGHT_KHZ:
+                sdlSampleRate = 8000;
+                break;
+            case SamplingRate.SIXTEEN_KHZ:
+                sdlSampleRate = 16000;
+                break;
+            case SamplingRate.TWENTY_TWO_KHZ:
+                // common sample rate is 22050, not 22000
+                // see https://en.wikipedia.org/wiki/Sampling_(signal_processing)#Audio_sampling
+                sdlSampleRate = 22050;
+                break;
+            case SamplingRate.FOURTY_FOUR_KHX:
+                // 2x 22050 is 44100
+                // see https://en.wikipedia.org/wiki/Sampling_(signal_processing)#Audio_sampling
+                sdlSampleRate = 44100;
+                break;
+        }
+
+        switch (sampleType) {
+            case BitsPerSample.EIGHT_BIT:
+                sdlSampleType = SampleType.UNSIGNED_8_BIT;
+                break;
+            case BitsPerSample.SIXTEEN_BIT:
+                sdlSampleType = SampleType.SIGNED_16_BIT;
+                break;
+        }
+    }
+
+    public void startAudioService(boolean encrypted) {
+        if (sdlInterface != null && sdlInterface.isConnected()) {
+            sdlInterface.addServiceListener(SessionType.PCM, this);
+            sdlInterface.startAudioService(encrypted);
+        }
+    }
+
+    public void stopAudioService() {
+        if (sdlInterface != null && sdlInterface.isConnected()) {
+            didRequestShutdown = true;
+            sdlInterface.stopAudioService();
+        }
+    }
+
+    public void pushAudioFile(File audioFile) {

```

Comment:
Change name to `play()` based on proposal https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0113-audiostreammanager.md#android

Reply:
The play method (with playImmediately) was proposed based on the experience with iOS that audio decoding only works through a file. The idea was to pre-decode and avoid delays. Fortunately we can live stream from an audio file which makes it irrelevant. Still the manager could benefint from some optimization (buffer limit)... working on it